frame/transaction-payment/src/lib.rs

```diff
@@ -792,4 +806,38 @@ mod tests {
 			assert_eq!(Balances::free_balance(2), 200 - 5 - 10 - 100 - 5);
 		});
 	}
+
+	// TODO Remove after u32 to u64 weights upgrade
+	#[test]
+	fn upgrade_to_fixed128_works() {
+		// TODO You can remove this from dev-dependencies after removing this test
+		use sp_storage::Storage;
+		use sp_runtime::Fixed64;
+		use frame_support::storage::generator::StorageValue;
+		use frame_support::traits::OnRuntimeUpgrade;
+		use core::num::NonZeroI128;
+
+		let mut s = Storage::default();
+
+		let original_multiplier = Fixed64::from_rational(1, 2);
+
+		let data = vec![
+			(
+				NextFeeMultiplier::storage_value_final_key().to_vec(),
+				original_multiplier.encode().to_vec()
+			),
+		];
+
+		s.top = data.into_iter().collect();
```

Comment:
what is this line doing?

Reply:
`data` is a vector of raw `(key, value)` pairs. This then converts it to a `BTreeMap`, which is expected for Storage `top`.